### PR TITLE
Dropdown: Remove delay for click events

### PR DESCRIPTION
**Pull request by ffxsam**
Please make sure these boxes are checked before submitting your PR, thank you!

* [X] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md)).
* [X] Make sure you are merging your commits to `dev` branch.
* [X] Add some descriptions and refer relative issues for you PR.

I noticed clicking on dropdowns was a bit laggy for some reason. Then I realized there was a delay being added (which is meant for hover). Having a timeout delay for clicks doesn't make any sense, and makes the UI feel sluggish.

### CODE REVIEW

File: packages/dropdown/src/dropdown.vue [hunk 1]
```diff
@@ -110,28 +110,32 @@
...
         }, this.showTimeout);
       },
-      hide() {
+      hide(options = {}) {
         if (this.triggerElm.disabled) return;
         this.removeTabindex();
         this.resetTabindex(this.triggerElm);
         clearTimeout(this.timeout);
+
+        const timeout = options.noTimeout ? 0 : this.hideTimeout;
```
Comment (Leopoldthecoder) [inline packages/dropdown/src/dropdown.vue]: `timeout` is defined but not used.

**Comment by Leopoldthecoder** [comment]
Thanks for contributing. I'll take care of this.

**Comment by ffxsam** [comment]
@Leopoldthecoder woops, dumb mistake because I rushed. I defined timeout but actually forgot to use it. 😉 But you can tell what I was trying to do.

**Comment by Leopoldthecoder** [comment]
Yes, so I did another PR #9573

**Comment by ffxsam** [comment]
Ahhh nice, I like your solution better. Thanks!

**Pull request closed without merging**